panda/src/audio/audioSound.h

```diff
@@ -88,15 +89,16 @@ class EXPCL_PANDA_AUDIO AudioSound : public TypedReferenceCount {
 
   // return: playing time in seconds.
   virtual PN_stdfloat length() const = 0;
-
+  
   // Controls the position of this sound's emitter.  px, py and pz are the
   // emitter's position.  vx, vy and vz are the emitter's velocity in UNITS
   // PER SECOND (default: meters).
-  virtual void set_3d_attributes(PN_stdfloat px, PN_stdfloat py, PN_stdfloat pz,
-                                 PN_stdfloat vx, PN_stdfloat vy, PN_stdfloat vz);
-  virtual void get_3d_attributes(PN_stdfloat *px, PN_stdfloat *py, PN_stdfloat *pz,
-                                 PN_stdfloat *vx, PN_stdfloat *vy, PN_stdfloat *vz);
+  virtual void set_3d_attributes(PN_stdfloat px, PN_stdfloat py, PN_stdfloat pz, PN_stdfloat vx, PN_stdfloat vy, PN_stdfloat vz);
+  virtual void get_3d_attributes(PN_stdfloat *px, PN_stdfloat *py, PN_stdfloat *pz, PN_stdfloat *vx, PN_stdfloat *vy, PN_stdfloat *vz);
 
+  // Controls the direction of this sound emitter. Currently implemented only for OpenAL.
+  virtual void set_3d_direction(LVector3 d);
+  virtual void get_3d_direction(LVector3 *d);
```

Comment:
I would just make this a return value, and mark it as `const`, ie. `virtual LVector3 get_3d_direction() const;`
Not sure why the other method was designed so strangely.